app/src/main/java/org/schabi/newpipe/player/ui/VideoPlayerUi.java

```diff
@@ -0,0 +1,1578 @@
+            animate(
+                    binding.playNextButton,
+                    show,
+                    duration,
+                    AnimationType.SCALE_AND_ALPHA);
+        }
+    }
+    //endregion
+
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // Repeat, shuffle, mute
+    //////////////////////////////////////////////////////////////////////////*/
+    //region Repeat, shuffle, mute
+
+    public void onRepeatClicked() {
+        if (DEBUG) {
+            Log.d(TAG, "onRepeatClicked() called");
+        }
+        player.cycleNextRepeatMode();
+    }
+
+    public void onShuffleClicked() {
+        if (DEBUG) {
+            Log.d(TAG, "onShuffleClicked() called");
+        }
+        player.toggleShuffleModeEnabled();
+    }
+
+    @Override
+    public void onRepeatModeChanged(@RepeatMode final int repeatMode) {
+        super.onRepeatModeChanged(repeatMode);
+
+        switch (repeatMode) {
+            case REPEAT_MODE_OFF:
+                binding.repeatButton.setImageResource(R.drawable.exo_controls_repeat_off);
+                break;
+            case REPEAT_MODE_ONE:
+                binding.repeatButton.setImageResource(R.drawable.exo_controls_repeat_one);
+                break;
+            case REPEAT_MODE_ALL:
+                binding.repeatButton.setImageResource(R.drawable.exo_controls_repeat_all);
+                break;
+            default:
+                break; // unreachable
+        }
+    }
+
+    @Override
+    public void onShuffleModeEnabledChanged(final boolean shuffleModeEnabled) {
+        super.onShuffleModeEnabledChanged(shuffleModeEnabled);
+        setShuffleButton(shuffleModeEnabled);
+    }
+
+    @Override
+    public void onMuteUnmuteChanged(final boolean isMuted) {
+        super.onMuteUnmuteChanged(isMuted);
+        setMuteButton(isMuted);
+    }
+
+    private void setMuteButton(final boolean isMuted) {
+        binding.switchMute.setImageDrawable(AppCompatResources.getDrawable(context, isMuted
+                ? R.drawable.ic_volume_off : R.drawable.ic_volume_up));
+    }
+
+    private void setShuffleButton(final boolean shuffled) {
+        binding.shuffleButton.setImageAlpha(shuffled ? 255 : 77);
+    }
+    //endregion
+
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // Other player listeners
+    //////////////////////////////////////////////////////////////////////////*/
+    //region Other player listeners
+
+    @Override
+    public void onPlaybackParametersChanged(@NonNull final PlaybackParameters playbackParameters) {
+        super.onPlaybackParametersChanged(playbackParameters);
+        binding.playbackSpeed.setText(formatSpeed(playbackParameters.speed));
+    }
+
+    @Override
+    public void onRenderedFirstFrame() {
+        super.onRenderedFirstFrame();
+        //TODO check if this causes black screen when switching to fullscreen
```

Comment:
TODO detected

Reply:
Unrelated to this PR: https://github.com/TeamNewPipe/NewPipe/blob/dev/app/src/main/java/org/schabi/newpipe/player/Player.java#L2584